src/parser.rs

```diff
@@ -322,24 +323,19 @@ impl Library {
     fn read_record_start(&mut self, parser: &mut Reader,
                          ns_id: u16, attrs: &Attributes) -> Result<()> {
 
-        let typ = try!(self.read_record(parser, ns_id, attrs));
-        let name = typ.get_name().clone();
-        self.add_type(ns_id, &name, typ);
+        if let Some(typ) = try!(self.read_record(parser, ns_id, attrs)) {
+            let name = typ.get_name().clone();
+            self.add_type(ns_id, &name, typ);
+        }
         Ok(())
     }
 
     fn read_record(&mut self, parser: &mut Reader, ns_id: u16, attrs: &Attributes)
-        -> Result<Type> {
-        let mut name = try!(
-            attrs
-                .by_name("name")
-                .ok_or_else(|| mk_error!("Missing record name", parser))
-        );
-        let mut c_type = try!(
-            attrs
-                .by_name("type")
-                .ok_or_else(|| mk_error!("Missing c:type attribute", parser))
-        );
+        -> Result<Option<Type>> {
+        let mut name = try!(attrs.by_name("name")
+                .ok_or_else(|| mk_error!("Missing record name", parser)));
+        let mut c_type = try!(attrs.by_name("type")
+                .ok_or_else(|| mk_error!("Missing c:type attribute", parser)));
```

Comment:
[rustfmt-nigthly](https://crates.io/crates/rustfmt-nightly) change this 2 variables back.